pkg/apis/autoscaling/validation/validation.go

```diff
@@ -337,13 +365,39 @@ func validateMetricSpec(spec autoscaling.MetricSpec, fldPath *field.Path) field.
 	return allErrs
 }
 
-func validateObjectSource(src *autoscaling.ObjectMetricSource, fldPath *field.Path) field.ErrorList {
+func validateObjectSource(src *autoscaling.ObjectMetricSource, fldPath *field.Path, opts HPAValidationOptions) field.ErrorList {
 	allErrs := field.ErrorList{}
 
 	allErrs = append(allErrs, ValidateCrossVersionObjectReference(src.DescribedObject, fldPath.Child("describedObject"))...)
 	allErrs = append(allErrs, validateMetricIdentifier(src.Metric, fldPath.Child("metric"))...)
 	allErrs = append(allErrs, validateMetricTarget(src.Target, fldPath.Child("target"))...)
 
+	if !opts.AllowMismatchedTargetTypeAndValue {
+		switch src.Target.Type {
+		case autoscaling.UtilizationMetricType:
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("target").Child("type"), src.Target.Type, "must not set Utilization type for Object source type"))
+		case autoscaling.ValueMetricType:
+			if src.Target.Value == nil {
+				allErrs = append(allErrs, field.Required(fldPath.Child("target").Child("value"), "must set target value for value metric type"))
+			}
+			if src.Target.AverageValue != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("target").Child("averageValue"), src.Target.AverageValue, "must not set target averageValue for value metric type"))
+			}
+			if src.Target.AverageUtilization != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("target").Child("averageUtilization"), src.Target.AverageUtilization, "must not set target averageUtilization for value metric type"))
+			}
```

Comment:
Drop the AverageValue / AverageUtilization checks here... these go further than the controller does and can fail an existing object that is currently working:

```yaml
object:
  ...
  target:
    type: Value
    value: "..."
    averageValue: "..."
    averageUtilization: "..."
```

That would pass validation today, work successfully with the controller today (averageValue and averageUtilization would be ignored), result in AllowMismatchedTargetTypeAndValue=false, and then get rejected on update by validation, which is not what we want.